lib/ReportDataCollector.php

```diff
@@ -460,71 +400,19 @@ private function getOCTablesArray() {
 		return $ocTables;
 	}
 
-	/**
-	 * @return array
-	 */
-	private function getPhpInfoDetailArray() {
-		$sensitiveServerConfigs = [
-			'HTTP_COOKIE',
-			'PATH',
-			'Cookie',
-			'include_path',
-		];
-
-		// Get the phpinfo, parse it, and record it (parts from http://www.php.net/manual/en/function.phpinfo.php#87463)
-		\ob_start();
-		\phpinfo(INFO_ALL & ~INFO_ENVIRONMENT);
-
-		$phpinfo = \preg_replace(
-			['#^.*<body>(.*)</body>.*$#ms', '#<h2>PHP License</h2>.*$#ms',
-				'#<h1>Configuration</h1>#', "#\r?\n#", "#</(h1|h2|h3|tr)>#", '# +<#',
-				"#[ \t]+#", '#&nbsp;#', '#  +#', '# class=".*?"#', '%&#039;%',
-				'#<tr>(?:.*?)" src="(?:.*?)=(.*?)" alt="PHP Logo" /></a>'
-				. '<h1>PHP Version (.*?)</h1>(?:\n+?)</td></tr>#',
-				'#<h1><a href="(?:.*?)\?=(.*?)">PHP Credits</a></h1>#',
-				'#<tr>(?:.*?)" src="(?:.*?)=(.*?)"(?:.*?)Zend Engine (.*?),(?:.*?)</tr>#',
-				"# +#", '#<tr>#', '#</tr>#'],
-			['$1', '', '', '', '</$1>' . "\n", '<', ' ', ' ', ' ', '', ' ',
-				'<h2>PHP Configuration</h2>' . "\n" . '<tr><td>PHP Version</td><td>$2</td></tr>' .
-				"\n" . '<tr><td>PHP Egg</td><td>$1</td></tr>',
-				'<tr><td>PHP Credits Egg</td><td>$1</td></tr>',
-				'<tr><td>Zend Engine</td><td>$2</td></tr>' . "\n" .
-				'<tr><td>Zend Egg</td><td>$1</td></tr>', ' ', '%S%', '%E%'],
-			\ob_get_clean()
-		);
-
-		$sections = \explode('<h2>', \strip_tags($phpinfo, '<h2><th><td>'));
-		unset($sections[0]);
-
-		$result = [];
-		$sensitiveServerConfigs = \array_flip($sensitiveServerConfigs);
-		foreach ($sections as $section) {
-			$n = \substr($section, 0, \strpos($section, '</h2>'));
-			if ($n === 'PHP Variables') {
-				continue;
-			}
-			\preg_match_all(
-				'#%S%(?:<td>(.*?)</td>)?(?:<td>(.*?)</td>)?(?:<td>(.*?)</td>)?%E%#',
-				$section,
-				$matches,
-				PREG_SET_ORDER
-			);
-			foreach ($matches as $match) {
-				if (isset($sensitiveServerConfigs[$match[1]])) {
-					continue;
-					// filter all key which contain 'password'
-				}
-				if (!isset($match[3])) {
-					$value = isset($match[2]) ? $match[2] : null;
-				} elseif ($match[2] == $match[3]) {
-					$value = $match[2];
-				} else {
-					$value = \array_slice($match, 2);
-				}
-				$result[$n][$match[1]] = $value;
-			}
+	private function getLinuxDistribution(): string {
+		if (file_exists('/etc/os-release')) {
+			return file_get_contents('/etc/os-release');
```

Comment:
do we need any safety check here (and for the "/etc/lsb-release")? I guess it's fine to assume the files haven't been tampered anyhow. I mean, if someone has modified those files he could also deal greater damage in other places instead of trying to do some kind of injection from those files.

Reply:
I guess we can assume that system config files are not corrupted.
Alternative approach: try to parse the files ...

I will anyhow move this to a different PR anyhow.